features/reset/reset_and_delete/feature.go

```diff
@@ -127,18 +125,24 @@ func CheckResult(ctx context.Context, r *harness.Runner, run client.WorkflowRun)
 	if next.GetEventType() != enums.EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED {
 		return errors.New("expected last event to be terminated")
 	}
-	// Ensure original run is findable via visibility & has correct status
-	resp, err := r.Client.ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
-		Namespace: r.Namespace,
-		Query:     "WorkflowId = '" + origRun.GetID() + "'",
-	})
-	if err != nil {
-		return err
-	}
-	r.Require.Equal(1, len(resp.GetExecutions()))
-	r.Require.Equal(enums.WORKFLOW_EXECUTION_STATUS_TERMINATED, resp.GetExecutions()[0].Status)
-
-	return err
+	// Use eventually since visibility is eventually consistent.
+	r.Require.EventuallyWithT(func(t *assert.CollectT) {
+		// Ensure original run is findable via visibility & has correct status
+		resp, err := r.Client.ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+			Namespace: r.Namespace,
+			Query:     "WorkflowId = '" + origRun.GetID() + "'",
+		})
+		assert.NoError(t, err)
+		if err != nil {
+			return
+		}
+		assert.Len(t, resp.GetExecutions(), 1)
+		if len(resp.GetExecutions()) != 1 {
+			return
+		}
+		assert.Equal(t, enums.WORKFLOW_EXECUTION_STATUS_TERMINATED, resp.GetExecutions()[0].Status)
+	}, 1*time.Second, 1*time.Minute)
```

Comment:
Can we speed this up, say to:

```suggestion
	}, 200*time.Millisecond, 10*time.Second)
```